org/postgresql/jdbc2/TypeInfoCache.java

```diff
@@ -150,9 +153,14 @@ public synchronized void addCoreType(String pgTypeName, Integer oid, Integer sql
         Character delim = ',';
         _arrayOidToDelimiter.put(oid, delim);
 
-        String pgArrayTypeName = "_" + pgTypeName;
```

Comment:
Easy fix seems to don't cache mapping if  

    boolean isArray = pgTypeName.endsWith("[]");

evaluated to true.
I agree that int4[] in output should be removed so it's consistent with type lookups.